streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -76,6 +76,9 @@ public void resize(final long newCacheSizeBytes) {
         final boolean shrink = newCacheSizeBytes < maxCacheSizeBytes;
         maxCacheSizeBytes = newCacheSizeBytes;
         if (shrink) {
+            if (caches.values().isEmpty()) {
```

Comment:
Why this check?

Reply:
Apparently `CircularIterators` throw an error if they are made on empty lists. And if there are no caches to resize we don't need to bother with the rest of this.

Reply:
I see -- so it's a bug fix on the side (not directly related to this RP) -- the original PR that added this method should have added this check.

Reply:
yeah, I didn't realize this was a problem, but when I added more test coverage it showed up